upup/pkg/fi/cloudup/new_cluster.go

```diff
@@ -679,7 +679,9 @@ func setupMasters(opt *NewClusterOptions, cluster *api.Cluster, zoneToSubnetMap
 				k8sVersion, err := version.ParseKubernetesVersion(cluster.Spec.KubernetesVersion)
 				if err == nil && version.IsKubernetesGTE("1.18", *k8sVersion) {
 					if g.Spec.InstanceMetadata == nil {
-						g.Spec.InstanceMetadata = &api.InstanceMetadataOptions{}
+						g.Spec.InstanceMetadata = &api.InstanceMetadataOptions{
+							HTTPPutResponseHopLimit: fi.Int64(2),
+						}
```

Comment:
Should check if HTTPPutResponseHopLimit is already set by user. Probably all this logic has to be updated a bit.

Reply:
Also, should be IMDSv2, if I'm not mistaken.

Reply:
I only set hop limit if InstanceMetadata is nil, so user should not have set anything. IMDSv2 is used when `httpTokens: required` and this is set ... somewhere else.

Reply:
`IMDSv2` meant in title.
Here there is a case when HTTPTokens can be set, but HTTPPutResponseHopLimit won't be set. Would not happen in practice but code should be consistent.

Reply:
The goal of this one was just to see if we can get the tests passing. We need to add more logic here soon to get CNI (well cilium anyway) working.